python/tank/context.py

```diff
@@ -1531,22 +1578,19 @@ def context_yaml_constructor(loader, node):
     """
     # lazy load this to avoid cyclic dependencies
     from .api import Tank
-    
+
     # get the dict from yaml
-    context_constructor_dict = loader.construct_mapping(node)
-    
+    context_constructor_dict = loader.construct_mapping(node, deep=True)
```

Comment:
Yaml Serialization/unserialization was half-broken. I discovered it when I added the additional_entities value to the Context serialization unit test.

Now it works.